### PR TITLE
[Feature] Support none-pk table replication from another cluster for olap table and cloud-native table

### PR DESCRIPTION
**Pull request by xiangguangyxg**
Why I'm doing:
To support starrocks cluster migration, from shared-nothing cluster to shared-nothing cluster or shared-data cluster.

What I'm doing:
Implement a replication job to support none-pk table replication from another source starrocks cluster

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by miomiocat — APPROVED** [review]

**Review by miomiocat — DISMISSED** [review]
LGTM!

It works in our environment~

File: be/src/storage/lake/replication_txn_manager.h [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+#include "storage/olap_common.h"
+#include "storage/olap_define.h"
+#include "storage/rowset/rowset_meta.h"
+
+namespace starrocks::lake {
+
+class ReplicationTxnManager {
+public:
+    ReplicationTxnManager(lake::TabletManager* tablet_manager) : _tablet_manager(tablet_manager) {}
+
```
Comment (sduzh) [inline be/src/storage/lake/replication_txn_manager.h:31]: DISALLOW_COPY_AND_MOVE(ReplicationTxnManager);

File: be/src/storage/lake/tablet.cpp [hunk 2]
```diff
@@ -74,10 +74,18 @@ Status Tablet::put_txn_log(const TxnLogPtr& log) {
     return _mgr->put_txn_log(log);
 }
 
+Status Tablet::put_txn_slog(const TxnLogPtr& log) {
+    return _mgr->put_txn_slog(std::move(log));
```
[the remark this answers is not in the document]
Comment (xiangguangyxg) [reply be/src/storage/lake/tablet.cpp]: done

File: be/src/storage/lake/replication_txn_manager.h [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+#include "storage/lake/types_fwd.h"
+#include "storage/olap_common.h"
+#include "storage/olap_define.h"
+#include "storage/rowset/rowset_meta.h"
+
+namespace starrocks::lake {
+
+class ReplicationTxnManager {
+public:
+    ReplicationTxnManager(lake::TabletManager* tablet_manager) : _tablet_manager(tablet_manager) {}
```
[the remark this answers is not in the document]
Comment (xiangguangyxg) [reply be/src/storage/lake/replication_txn_manager.h]: done

File: be/src/storage/lake/replication_txn_manager.h (hunk 1 above)
Comment (xiangguangyxg) [reply be/src/storage/lake/replication_txn_manager.h:31]: done

**Review by sduzh — DISMISSED** [review]

**Review by sduzh — APPROVED** [review]

**Review by miomiocat — APPROVED** [review]

**Comment by xiangguangyxg** [comment]
@Mergify backport branch-3.2 branch-3.1

**Comment by xiangguangyxg** [comment]
backport 3.2 pr: https://github.com/StarRocks/starrocks/pull/38133
backport 3.1 pr: https://github.com/StarRocks/starrocks/pull/38208

**Pull request merged**